src/test/java/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,95 @@
+    public static final int BANANA_QUANTITY = 130;
+    public static final int MANGO_QUANTITY = 100;
+    public static final int KIWI_QUANTITY = 80;
+    public static final String BANANA = "banana";
+    public static final String APPLE = "apple";
+    public static final String MANGO = "mango";
+    public static final String KIWI = "kiwi";
+    private static Map<FruitTransaction.Operation, OperationHandler>
+            operationHandlerMap = new HashMap<>();
+    private static OperationStrategy operationStrategy;
+    private static FruitShopService fruitShopService;
+
+    @BeforeClass
+    public static void beforeAll() {
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE,
+                new BalanceOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN,
+                new ReturnOperationHandler());
+        operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+        fruitShopService = new FruitShopServiceImpl(operationStrategy);
+
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        Storage.fruits.clear();
+    }
+
+    @Test
+    public void calculate_validData_ok() {
+        List<FruitTransaction> expected = List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 100),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 100),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "kiwi", 100),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "mango", 100),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 30),
+                new FruitTransaction(FruitTransaction.Operation.RETURN, "apple", 20),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "kiwi", 20));
+        fruitShopService.calculate(expected);
+        int actualMango = Storage.fruits.get(MANGO);
+        Assert.assertEquals(
+                "FruitShopService failed or some problem with BalanceOperationHandler",
+                MANGO_QUANTITY, actualMango);
+        int actualApple = Storage.fruits.get(APPLE);
+        Assert.assertEquals(
+                "FruitShopService failed or some problem with ReturnOperationHandler",
+                APPLE_QUANTITY, actualApple);
+        int actualBanana = Storage.fruits.get(BANANA);
+        Assert.assertEquals(
+                "FruitShopService failed or some problem with SupplyOperationHandler",
+                BANANA_QUANTITY, actualBanana);
+        int actualKiwi = Storage.fruits.get(KIWI);
+        Assert.assertEquals(
+                "FruitShopService failed or some problem with PurchaseOperationHandler",
```

Comment:
to avoid the situation with some not identified handler problem - we need to add separate tests of strategy and its separate tests for each handler